### PR TITLE
Initial support for running newer versions of Django

### PR DESCRIPTION
**Pull request by dcramer**
This adds some transitionary components which should retain compatibility in 1.6. The goal is to get the codebase running in both 1.6 and at least 1.7 so the switchover is easier.

#nochanges

### CODE REVIEW

File: src/sentry/api/endpoints/event_apple_crash_report.py [hunk 1]
```diff
@@ -2,7 +2,13 @@
 
 import six
 
-from django.http import HttpResponse, CompatibleStreamingHttpResponse
+try:
```
Comment (dcramer) [inline src/sentry/api/endpoints/event_apple_crash_report.py]: im a bit concerned about this, but i think its possibly ok due to changes in middleware behavior

**Comment by dcramer** [comment]
1.8 is going to be tricky, but its close for 1.7:

- BitField needs fixed (current package has broken setup.py)
- TypeError: Database is trying to update a relational field of type BoundedIntegerField with a value of type ScoreClause. Make sure you are setting the correct relations

**Review by mattrobenolt — APPROVED** [review]
wcgw

**Comment by dcramer** [comment]
Going to merge this once green (I updated it to the latest encryption changes for South compat)

**Review by mitsuhiko — APPROVED** [review]
Seems okay. Not sure though about other parts of the codebase not in this repo.

**Comment by dcramer** [comment]
@mattrobenolt @mitsuhiko mind skimming this over again? i made a few more changes -- mostly vendoring bitfield

**Review by mattrobenolt — APPROVED** [review]
yolo

**Pull request merged**